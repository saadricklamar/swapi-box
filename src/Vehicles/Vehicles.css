.cardsContainer {
    position: absolute;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
    grid-auto-rows: auto;
    grid-gap: 1rem;
    width: 85%;
    margin: 0 auto;
    padding-bottom: 10px;
    margin-top: -1350px;
    margin-left: 250px;
}

.vehicles {
    height: 373px;
    width: 250px;
    margin: 5px;
    background-color: transparent;
    border-radius: 5px;
    display: flex;
    flex-wrap: wrap;
    text-align: left;
    font-size: 12px;
    border: 2px solid gold;
  }

h2 {
    margin: 0 10px 0 0;
    padding: 0 40px 0 5px;
    color: white;
    font-weight: bold;
    font-family: 'Cinzel', serif;
}

img {
  width: 250px;
  height: 160px;
  margin: 0;
  background: linear-gradient( rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2) ); 
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  border-bottom: 1px solid orange;
  
}

.fa-star{
  color: white;
  font-size: 26px;
}

.fa-star:hover{
  cursor: pointer;
  color: white;
}
