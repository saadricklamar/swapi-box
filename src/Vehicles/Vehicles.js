import React from 'react';
import './Vehicles.css';


const Vehicles = (props) => {
  const cardInfo = props.vehicles.map(card => {
    let vehicleName = card.name;
      return  <article className='card'>
                <img src={require(`../images/${vehicleName}.jpeg`)}/>
                <i className='fas fa-star'></i> 
                <div className='dark-background'>
                  <h2 className='name'>{card.name}</h2>
                  <h2>Model: {card.model}</h2>
                  <h2>Class: {card.vehicle_class}</h2>
                  <h2>Passengers: {card.passengers}</h2>
                </div>
            </article>
  })
    return (
    <div className='cardsContainer'>
     {cardInfo}
    </div>
  
    );
  
}

export default Vehicles;