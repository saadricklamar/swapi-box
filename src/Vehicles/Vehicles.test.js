import React from 'react';
import Vehicles from './Vehicles.js';
import { shallow } from 'enzyme';


const mockData = [];

describe('WelcomeHome', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Vehicles
                      vehicles={mockData}
                      />)
});
it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
});