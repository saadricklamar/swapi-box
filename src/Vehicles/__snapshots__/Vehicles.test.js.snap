// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeHome Should match the snapshot 1`] = `
<div
  className="cardsContainer"
/>
`;
